### PR TITLE
Use new filtering API for existing filter of synthetic methods

### PR DESCRIPTION
**Pull request by Godin**

### CODE REVIEW

File: org.jacoco.core/src/org/jacoco/core/internal/analysis/filter/SyntheticFilter.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+ *    Evgeny Mandrikov - initial API and implementation
+ *
+ *******************************************************************************/
+package org.jacoco.core.internal.analysis.filter;
+
+import org.objectweb.asm.Opcodes;
+import org.objectweb.asm.tree.MethodNode;
+
+/**
+ * Filters synthetic methods.
```
Comment (marchof) [inline org.jacoco.core/src/org/jacoco/core/internal/analysis/filter/SyntheticFilter.java]: Maybe we should be more specific here: "Filters synthetic methods unless they represent desugared lambda bodies."

File: org.jacoco.core/src/org/jacoco/core/internal/analysis/ClassAnalyzer.java [hunk 2]
```diff
@@ -102,8 +102,7 @@ private boolean isMethodFiltered(final String className,
 				return true;
 			}
 		}
-		return (access & Opcodes.ACC_SYNTHETIC) != 0
```
Comment (marchof) [inline org.jacoco.core/src/org/jacoco/core/internal/analysis/ClassAnalyzer.java]: Deleting these lines needs more cleanup to get rid of compiler warnings:

- `import org.objectweb.asm.Opcodes;` is unused and should be deleted
- Method parameter `access` is unused and should be removed

File: org.jacoco.core/src/org/jacoco/core/internal/analysis/filter/SyntheticFilter.java (hunk 1 above)
Comment (Godin) [inline org.jacoco.core/src/org/jacoco/core/internal/analysis/filter/SyntheticFilter.java]: @marchof fixed

File: org.jacoco.core/src/org/jacoco/core/internal/analysis/ClassAnalyzer.java (hunk 2 above)
Comment (Godin) [inline org.jacoco.core/src/org/jacoco/core/internal/analysis/ClassAnalyzer.java]: @marchof fixed

**Pull request merged**